### ISSUE TITLE
Settings lost

### ISSUE THREAD

**Issue by jrial** [atom/atom#14909]
### Prerequisites

* [ ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

Sometimes when updating packages, Atom loses most of its settings. E.g. right now, I updated Hyperclick, and after the restart I was greeted with the welcome message, my font was reset to the default settings, my settings for linter-pycodestyle and linter-pylint were lost, and I assume all the others as well, but I didn't bother with the rest much, so no impact there in my case.

This is difficult to reproduce; I can update packages dozens of times, but every once in a while, it loses all my settings. It's not a disk full issue, as I have 56GB available.

The data loss appears to be limited to the things in ~/.atom/config.cson - the other .cson files and styles.less appear to be unaffected.

### Steps to Reproduce

1. Update packages
2. Restart

**Expected behavior:** Atom restarts with upgraded packages, and settings intact.

**Actual behavior:** Atom restarts with upgraded packages, and lost settings.

**Reproduces how often:** Less than 10%

### Versions
Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
apm  1.18.1
npm  3.10.10
node 6.9.5 x64
python 2.7.6
git 1.9.1

OS: Linux Mint 17.3 Rosa
Kernel: 3.19.0-32-generic #37~14.04.1-Ubuntu SMP Thu Oct 22 09:41:40 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by thomasjo** [comment]
It would be really great if you could provide us with a list of installed packages. I can almost guarantee that this is caused by an installed community package, and if we know which packages you have installed, we can likely narrow it down to the culprit.

The simplest way is to execute `apm ls -i` from a terminal.

**Comment by jrial** [comment]
```
Community Packages (43) /home/nighty/.atom/packages
├── Zen@0.18.0
├── atom-beautify@0.30.3
├── atom-cli-diff@1.9.0
├── atom-django@0.3.2
├── atom-meld@0.3.5
├── atom-python-virtualenv@0.14.0
├── atom-transpose@0.3.4
├── autocomplete-python@1.9.1
├── blame@0.12.0
├── busy-signal@1.4.3
├── custom-title@1.0.1
├── data-atom@0.25.1
├── djangorad@0.0.1
├── emmet@2.4.3
├── emmet-snippets-compatibility@1.0.6
├── file-watcher@1.1.0
├── git-plus@7.9.3
├── git-time-machine@1.5.9
├── hyperclick@0.1.1
├── indent-guide-improved@1.4.12
├── intentions@1.1.2
├── language-django@0.1.1
├── language-nginx@0.8.0
├── linter@2.2.0
├── linter-pycodestyle@2.0.2
├── linter-pylint@2.1.0
├── linter-ui-default@1.6.2
├── local-history@4.3.0
├── merge-conflicts@1.4.5
├── minimap@4.28.2
├── minimap-linter@2.0.0
├── monokai@0.24.0
├── open-html-in-browser@0.1.0
├── open-terminal-here@2.3.0
├── persistent-undo@1.2.0
├── project-manager@3.3.5
├── python-tools@0.6.9
├── selection-highlight@0.1.6
├── sort-lines@0.14.0
├── symbols-tree-view@0.14.0
├── tag@0.5.0
├── termination@0.5.1
└── up2date@0.1.2
```

**Comment by bcroq** [comment]
Happened to me today after updating [python-indent](https://atom.io/packages/python-indent) to 1.1.0

**Comment by scb00** [comment]
This happened to me too. I just opened Atom this morning and all settings were gone (this includes core and theme settings, and also package settings). Style.less is unaffected. Key bindings are ok too. These are my packages, in case it helps.

```
Community Packages (32) /home/sergio/.atom/packages
├── atom-beautify@0.30.3
├── atom-ternjs@0.18.3
├── atom-typescript@11.0.6
├── autocomplete-paths@1.0.5
├── autocomplete-php@0.3.7
├── autocomplete-sass@0.1.0
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── cursor-history@0.11.1
├── docblockr@0.10.5
├── drupal@1.9.0
├── file-type-icons@1.3.4
├── git-history@3.3.0
├── highlight-selected@0.13.1
├── intentions@1.1.2
├── jquery-snippets@11.0.0
├── language-diff@0.7.0
├── line-diff-details@1.8.0
├── linter@2.2.0
├── linter-eslint@8.2.1
├── linter-flake8@2.2.1
├── linter-php@1.3.2
├── linter-sass-lint@1.7.4
├── linter-scss-lint@3.1.0
├── linter-ui-default@1.6.2
├── merge-conflicts@1.4.5
├── pigments@0.39.1
├── platformio-ide-terminal@2.5.3
├── remote-sync@4.1.5
├── show-in-system@0.4.0
├── split-diff@1.4.3
└── synced-sidebar@0.4.3
```

**Comment by rsese** [comment]
Closed https://github.com/atom/atom/issues/14922 as a dupe of this one, copying over some comments with package lists and other details.

*From @alexeypetrushin*

not reproducible

```
Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0
```

Mac OS 10.12.4

Have no idea when it updates, I haven't updated it manually

I restarted it

```
├── atom-typescript@11.0.3
├── atom-typescript-beta@11.0.1
├── autocomplete-elixir@1.5.14
├── busy-signal@1.4.3
├── intentions@1.1.2
├── language-elixir@0.19.0
├── language-julia@0.11.0
├── language-matlab@0.2.1
├── latex-completions@0.3.2
├── linter@2.2.0
├── linter-elixirc@1.6.0
├── linter-ui-default@1.6.2
└── toggle-gutter@0.3.3
```

**Comment by rsese** [comment]
*From @runartrollet:*

Happened to me too today. Atom was acting a bit slow, which it sometimes does at my old computer. Restarting Atom usually helps, so I did just that and my settings from `config.cson` were gone. 
I am running Atom 1.18 on macOS Sierra.

Installed packages: 
```
├── Sublime-Style-Column-Selection@1.7.4
├── aligner-python@1.1.0
├── atom-alignment@0.13.0
├── atom-beautify@0.30.3
├── atom-clock@0.1.12
├── atom-css-clean@2.21.0
├── atom-dedupe-open-files@0.1.0
├── atom-like-brackets-editor@1.2.0
├── atom-reveal-file-in-finder@1.1.4
├── atom-runner@2.7.1
├── atom-typescript@11.0.6
├── autocomplete-css@0.16.2
├── autocomplete-html@0.8.0
├── autocomplete-json@5.5.0
├── autocomplete-python@1.9.1
├── autoprefixer@3.7.1
├── browser-plus@0.0.97
├── browser-refresh-on-save@0.0.6
├── build-npm-apm@0.12.0
├── busy-signal@1.4.3
├── change-case@0.6.5
├── color-picker@2.2.5
├── colorblind-wombat-dark-syntax@0.6.0
├── cursor-history@0.11.1
├── dash@1.7.1
├── emmet@2.4.3
├── enhanced-tabs@1.7.0
├── expand-region@0.3.0
├── expand-selection-to-quotes@0.11.0
├── file-icons@2.1.8
├── filecolor@0.4.0
├── fold-comments@0.6.0
├── gitignore-snippets@0.2.5
├── golden-ratio@0.3.0
├── goto@1.8.3
├── highlight-line@0.12.0
├── highlight-selected@0.13.1
├── intentions@1.1.2
├── javascript-snippets@1.2.1
├── jumpy@3.1.3
├── kite@0.30.1
├── language-c@0.58.1
├── language-clojure@0.22.3
├── language-csharp@0.14.2
├── language-css@0.42.2
├── language-csv@1.1.2
├── language-django@0.1.1
├── language-gfm@0.89.1
├── language-go@0.44.1
├── language-html@0.47.3
├── language-java@0.27.2
├── language-javascript@0.126.1
├── language-json@0.19.1
├── language-json-comments@0.16.3
├── language-latex@1.0.0
├── language-less@0.32.0
├── language-lua@0.9.11
├── language-make@0.22.3
├── language-markdown@0.23.0
├── language-mustache@0.14.1
├── language-objective-c@0.15.1
├── language-perl@0.37.0
├── language-pgsql@0.2.3
├── language-php@0.39.0
├── language-python@0.45.3
├── language-restructuredtext@1.1.0
├── language-sass@0.59.0
├── language-sql@0.25.6
├── language-sql-mysql@0.3.1
├── language-xml@0.35.1
├── language-yaml@0.30.0
├── latex-hyperclick@0.2.0
├── latextools@0.8.5
├── linter@2.2.0
├── linter-chktex@1.3.1
├── linter-csslint@1.3.4
├── linter-htmlhint@1.3.3
├── linter-jshint@3.1.5
├── linter-jsonlint@1.3.0
├── linter-pylama@0.9.3
├── linter-python-pep8@0.2.0
├── linter-rst@0.0.3
├── linter-spell@0.14.0
├── linter-spell-latex@0.9.2
├── linter-ui-default@1.6.2
├── lipsum@0.1.0
├── lorem@1.2.1
├── markdown-preview-plus@2.4.10
├── markdown-writer@2.6.5
├── open-in-browsers@0.0.30
├── open-recent@5.0.0
├── pdf-view@0.59.0
├── pigments@0.39.1
├── preview-inline@1.4.7
├── python-indent@1.1.0
├── remove-whitespace@0.2.1
├── rulerz@0.4.3
├── script@3.15.0
├── simply-django-atom@0.0.2
├── sort-lines@0.14.0
├── spell-check@0.72.0
├── split-diff@1.4.3
├── swackets@0.26.0
├── symbols-view@0.116.1
├── sync-settings@0.8.2
├── tab-to-escape@0.1.0
├── tablr@1.8.3
├── tabularize@0.2.5
├── text-pastry@0.6.1
├── todo-show@1.11.0
├── turbo-javascript@1.4.0
├── wakatime@7.0.3
└── zentabs@0.8.8
```

**Comment by rsese** [comment]
*From @mminkoff:*

It's happened to me twice in the last day.  Once was in the middle of a session.

Atom    : 1.18.0
Electron: 1.3.15
Chrome  : 52.0.2743.82
Node    : 6.5.0

macOS Sierra 10.12.5

├── Handlebars@1.8.0
├── advanced-open-file
├── aligner@1.2.3
├── aligner-javascript@1.2.0
├── atom-beautify@0.30.4
├── atom-ember-frost-syntax@1.0.0
├── atomic-emacs@0.11.0
├── autoclose-html@0.23.0
├── autocomplete-clang@0.11.3
├── autocomplete-emojis@2.5.0
├── build@0.68.0
├── busy@0.7.0
├── busy-signal@1.4.3
├── case-keep-replace@0.6.0
├── chestnut-light-atom-syntax@0.2.1
├── clipboard-plus@0.5.1
├── coffee-compile@0.26.0
├── color-picker@2.2.5
├── emacs-flow@0.1.2
├── emacs-plus@0.11.1
├── ember-cli-helper@0.9.0
├── ember-tabs@2.4.3
├── file-icons@2.1.8
├── git-blame@1.1.1
├── git-tab-status@1.9.2
├── gruvbox-plus-syntax@1.3.1
├── intentions@1.1.2
├── isotope-light-ui@2.2.7
├── language-ini@1.19.0
├── linter@2.2.0
├── linter-ember-template@0.2.0
├── linter-eslint@8.2.1
├── linter-gcc@0.7.1
├── linter-js-standard@3.9.1
├── linter-jshint@3.1.5
├── linter-ui-default@1.6.2
├── merge-conflicts@1.4.5
├── minimap@4.28.3
├── node-debugger@1.10.1
├── package-list@0.1.2
├── pigments@0.39.1
├── platformio-ide@2.0.0-beta.7
├── platformio-ide-debugger@1.2.3
├── platformio-ide-terminal@2.5.3
├── prettier-atom@0.36.1
├── project-manager@3.3.5
├── sequential-command@0.3.0
├── solarized-dark-apparent-ui-successor@0.2.0
├── solarized-dark-ui@0.3.2
├── solarized-light-ui@0.1.0
├── split-diff@1.4.3
├── text-transformer@0.1.1
├── tool-bar@1.1.0
├── unicorn-light-syntax@3.1.0
├── unicorn-light-ui@2.1.0
├── unicorn-ui@1.2.0
├── zenburn@3.1.0
└── zentabs@0.8.8

though not all packages are enabled.

1.18 was installed some time ago.  One of my packages might have been recently installed but I don't remember which.  When it happened today I was in the middle of editing a file.

---

Packages I have in common with @runartrollet:

├── busy-signal@1.4.3 (disabled for me)
├── color-picker@2.2.5 
├── file-icons@2.1.8
├── intentions@1.1.2
├── linter@2.2.0
├── linter-jshint@3.1.5
├── linter-ui-default@1.6.2
├── pigments@0.39.1
├── split-diff@1.4.3
└── zentabs@0.8.8

None seem to have been updated in the last few days.

**Comment by rsese** [comment]
As mentioned in https://github.com/atom/atom/issues/14922#issuecomment-316853315, if anyone finds a reproducible case, please let us know since that would be a big help in tracking down the problem.

One question that came up - can anyone confirm if they happen to be storing their config file in Dropbox or something similar?

**Comment by 50Wliu** [comment]
I can confirm that my config.cson was just regenerated.  No, I am not storing it in Dropbox.

**Comment by jrial** [comment]
Not storing on Dropbox either. Running Atom in a VM though, in case it's relevant.

**Comment by geovanisouza92** [comment]
Mee too, I'd disabled some core packages too:

```
Community Packages (24) /home/geovani/.atom/packages
├── activate-power-mode@2.0.0
├── atom-beautify@0.30.4
├── busy-signal@1.4.3
├── ctrl-dir-scroll@0.2.3
├── editorconfig@2.2.2
├── electron-highlighter-syntax@1.9.0
├── emmet@2.4.3
├── file-icons@2.1.9
├── highlight-selected@0.13.1
├── intentions@1.1.2
├── language-diff@0.7.0
├── language-docker@1.1.8
├── linter@2.2.0
├── linter-js-standard@3.9.1
├── linter-js-standard-engine@2.1.0
├── linter-jsonlint@1.3.0
├── linter-rubocop@2.1.1
├── linter-ui-default@1.6.3
├── nyancat@0.14.0
├── prettier-atom@0.37.0
├── prettier-eslint@1.2.9
├── react@0.16.2
├── sort-lines@0.14.0
└── split-diff@1.4.3
```

```
"*":
  "activate-power-mode":
    plugins:
      playAudio: true
      screenShake: true
  "atom-beautify":
    xml:
      default_beautifier: "JS Beautify"
  core:
    autoHideMenuBar: true
    disabledPackages: [
      "activate-power-mode"
      "language-docker"
      "language-diff"
      "linter-js-standard-engine"
      "linter-rubocop"
      "autocomplete-atom-api"
      "autoflow"
      "dev-live-reload"
      "exception-reporting"
      "language-clojure"
      "language-coffee-script"
      "language-csharp"
      "language-java"
      "language-less"
      "language-make"
      "language-mustache"
      "language-objective-c"
      "language-perl"
      "language-php"
      "language-property-list"
      "language-sass"
      "language-toml"
      "markdown-preview"
      "metrics"
      "open-on-github"
      "package-generator"
      "styleguide"
      "welcome"
      "whitespace"
    ]
    telemetryConsent: "no"
    themes: [
      "one-dark-ui"
      "electron-highlighter-syntax"
    ]
  editor:
    fontFamily: "'Fira Code', 'Droid Sans Mono'"
    fontSize: 16
    scrollPastEnd: true
    showInvisibles: true
  "exception-reporting":
    userId: "ae1ba575-4901-42af-b4d7-64cd1cfce852"
  "linter-ui-default":
    panelHeight: 85
  "prettier-atom":
    formatOnSaveOptions: {}
    prettierOptions:
      semi: false
      singleQuote: true
    useEslint: true
  welcome:
    showOnStartup: false
  "xml-formatter":
    endLineCharacter: "LF (\\n)"
```

**Comment by ungb** [comment]
I'm unable to get a repro for this issue, but it seems like common packages that are installed are:

```
file-icons@2.1.8
linter@2.2.0
linter-ui-default@1.6.2
intentions@1.1.2
busy-signal@1.4.3
```


Based on the repro steps this occurs during updating packages.  Was there any thing else going on while Atom was being updated?  Maybe linter was running?  

I've attempted to install a bunch of packages listed above and updated all packages one by one and `update all` and haven't seen my config get reset.  

If any one can help provide more info or a steady repro that would be very helpful.   


In the meantime, it's a good idea to save your settings to git by doing the following:
1. Run the following: 
```bash
cd ~/.atom # or where your atom folder is
git init
```

2. Add the following to .gitignore in ~/.atom and save
```
.apm
.node-gyp
blob-store
compile-cache
# Only add your packages if you don't want your packages to be stored in git.
packages
snapshot-cache
storage
```

3. run `git status` to see what's changes are unstaged

4. Add your changes and commit to git local repo.
```
git add --all
git commit -am "Add initial config files in .atom folder"

```


alternatively, you can also you the [sync-settings](https://atom.io/packages/sync-settings) package.

**Comment by weijarz** [comment]
It's happened to me twice today. No packages been updated in the last few days.

Atom    : 1.18.0
Electron: 1.4.16
Chrome  : 53.0.2785.143
Node    : 6.5.0
OS: ArchLinux

```
Community Packages (26) /home/wjzhou/.atom/packages
├── atom-terminal@0.8.0
├── auto-indent@0.5.0
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── ctrl-dir-scroll@0.2.3
├── editorconfig@2.2.2
├── flow-ide@1.7.0
├── git-history@3.3.0
├── goto-definition@1.3.4
├── highlight-selected@0.13.1
├── hyperclick@0.1.3
├── indentation-indicator@1.0.0
├── intentions@1.1.5
├── language-ini@1.19.0
├── language-nginx@0.8.0
├── language-postcss@1.3.1
├── language-vue@0.23.0
├── last-cursor-position@0.9.2
├── linter@2.2.0
├── linter-eslint@8.2.1
├── linter-ui-default@1.6.3
├── open-recent@5.0.0
├── resize-indent@0.3.0
├── sort-lines@0.14.0
├── spacegray-black-syntax@0.1.4
└── zentabs@0.8.8
```

**Comment by Ben3eeE** [comment]
Happened to me today on Windows 7 Atom 1.19.2 while undoing/redoing in a JavaScript file.

**Comment by Ben3eeE** [comment]
Here is a what I did before running into this issue:

1. Went through my list of packages and enabled all that I want to use. I had a mess with packages because I have been running various setups and packages that I don't use to reproduce issues. I uninstalled several of those packages
1. Restarted Atom after enabling, disabling and uninstalling all packages
1. Editing normally like I would any day
1. Navigated to a JavaScript file that was already open in a tab by clicking on the tab. I wanted to remove two console log lines. I accidentally hit `ctrl-z` instead of `ctrl-x` twice
1. Tried to run redo to correct my mistake
1. After the first or second redo Atom was completely unresponsive for 5 minutes and went white. I left it running hoping I would catch some errors in the developer tools console if I left it running
1. When Atom was responsive again the developer tools showed errors from packages I have disabled. This is because they are not configured properly
1. I checked the settings view and notice that all my disabled packages are enabled and go through the list to disable them again
1. I restart Atom and I am presented with the welcome guide and notice that all my settings have been reset

**Comment by nathansobo** [comment]
One thought. We automatically re-read and evaluate settings when we detect a file change. I wonder if saving the settings to an invalid state when manually editing might be involved.

**Comment by jrial** [comment]
@nathansobo In my case, I can rule out this hypothesis. I don't often edit Atom's settings files, stylesheets nor anything.

However, I don't rule out that some plugin might do it either through its settings screen, or automatically in the background.

**Comment by nathansobo** [comment]
Thanks for the info @jrial. Really sorry this is happening.

**Comment by Ben3eeE** [comment]
@nathansobo I have only manually edited the settings file in a few rare cases when I had to and the last time was a couple of months ago.

To add to my information above I did change `Soft Wrap` to be on and and `Soft Wrap Hanging Indent`  from nothing to 4. I can't recall when I changed these settings but if I remember correctly it was after enabling, disabling and uninstalling packages before restarting Atom.

**Comment by nathansobo** [comment]
Maybe it has something to do with a weird timing issue when we're writing out the settings programmatically then. Thanks for ruling out my hypothesis.

**Comment by kathenas** [comment]
Just got hit by this bug running 1.19.2 (debian x64 packagr from atom.io).

philwyett@ks-yoda:$ apm ls -i
Community Packages (14) /home/philwyett/.atom/packages
├── atom-beautify@0.30.5
├── busy-signal@1.4.3
├── intentions@1.1.5
├── language-diff@0.7.0
├── linter@2.2.0
├── linter-clang@4.1.2
├── linter-markdown@5.2.0
├── linter-php@1.4.0
├── linter-pylint@2.1.0
├── linter-shellcheck@1.4.4
├── linter-stylelint@4.0.2
├── linter-ui-default@1.6.7
├── minimap@4.29.6
└── wordcount@2.10.4

I have now re-done my config and created a backup of it. :-)

I will now use atom, though this time I will run with no packages disabled and lets see if we hit the issue again.

Config:

"*":
  core:
    allowPendingPaneItems: false
    audioBeep: false
    automaticallyUpdate: false
    openEmptyEditorOnStart: false
    reopenProjectMenuCount: 0
    restorePreviousWindowsOnStart: "no"
    telemetryConsent: "no"
  editor:
    atomicSoftTabs: false
    fontSize: 12
    preferredLineLength: 79
    showIndentGuide: true
    tabLength: 4
    tabType: "soft"
    zoomFontWhenCtrlScrolling: false
  "exception-reporting":
    userId: "1a3de77e-a76f-4d40-9638-bc4b04fe3163"
  "linter-pylint":
    disableTimeout: true
  "linter-shellcheck":
    enableNotice: true
  "linter-stylelint":
    disableWhenNoConfig: false
    useStandard: true
  "linter-ui-default":
    showPanel: true
  minimap:
    displayPluginsControls: false
  welcome:
    showOnStartup: false
  whitespace:
    removeTrailingWhitespace: false

**Comment by derwaldgeist** [comment]
Same problem here. I updated two packages today, and after that Atom resetted some of my settings (e.g. the theme), but not all of them. It also shows the Welcome Guide etc. on *every* startup now, even if I close them and reload Atom. I just can't get rid of them, which is quite annoying. :-(

I'm running Atom 1.19.3 on macOS Sierra.

**Comment by yubaoquan** [comment]
Same problem here. Happened twice, the second time when it happened, I was adjusting font size by pressing cmd-+.

The only different operation I did today was installing the package hydrogen, and edit one of its setting options  `Kernel Specs`. 

### versions:
Atom    : 1.20.0-beta5
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

apm  1.18.4
npm  3.10.10
node 6.9.5 x64
python 3.6.1
git 2.11.0

OS: macOS Sierra Version 10.12.6

community packages
├── Hydrogen@1.20.0
├── atom-beautify@0.30.5
├── atom-material-syntax-dark@1.0.0
├── atom-material-ui@2.0.4
├── atom-terminal@0.8.0
├── auto-close-html2@0.3.8
├── backspace-until-not-whitespace@0.1.7
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── convert-to-utf8@0.2.4
├── dbclick-tree-view@1.6.0
├── docblockr@0.11.0
├── double-tag@0.9.0
├── emmet@2.4.3
├── file-types@0.5.5
├── highlight-selected@0.13.1
├── hyper-alien-ui-theme@0.4.0
├── hyper-ui@0.1.3
├── intentions@1.1.5
├── language-freemarker@0.2.0
├── language-lisp@0.2.0
├── language-pug@0.0.21
├── language-vue@0.23.1
├── linter@2.2.0
├── linter-eslint@8.2.1
├── linter-ui-default@1.6.8
├── minimap@4.29.6
├── minimap-highlight-selected@4.6.1
├── package-cop@0.2.10
├── seti-syntax@1.1.3
├── seti-ui@1.9.0
├── vim-mode-plus@1.0.0
├── vim-mode-plus-keymaps-for-surround@0.2.1
└── zentabs@0.8.8

**Comment by andrew-aladev** [comment]
Same issue. I couldn't use atom because of this issue.

Ok, we are updating packages and something in this packages could be broken during updates. But how it is possible to lost settings? Settings should not be linked with packages. Who is responsible for such settings design? It should be punished properly.

**Comment by rsese** [comment]
@andrew-aladev your comment was deleted as a violation of the [Atom Code of Conduct](https://github.com/atom/atom/blob/master/CODE_OF_CONDUCT.md) as it was insulting or derogatory. You may consider this an official warning.

**Comment by andrew-aladev** [comment]
@rsese I don't care. I see that settings system can reset all configs without any warning. It's hard for me to remember all config variables now. I can lost several days to remember them.

Please add warning to the `readme` that user should create a backup of `.atom` directory. It could be `cvs`, `cron + rsync`, etc. Please don't start to fix this issue before fixing readme. Thank you.

**Comment by smashwilson** [comment]
Okay, because we never did find a consistent way to repro this, we can't be certain this is really "closed" once and for all just yet, so I'm leaving the issue open. But, to me, this really has the scent of a filesystem race condition.

@kuychaco and I went over the `Config` source with a fine-toothed comb and identified three race conditions that _could_ be the culprit. The most likely fix is that we tightened up some of the sequencing that could result in the configuration being written before the initial load from disk has completed, which would result in an empty config overwriting the real one.

**Comment by focusaurus** [comment]
Happened to me today when restarting after upgrading to atom 1.20.0. Note I have `~/.atom/config.cson` as a symlink to `/Users/plyons/projects/dotfiles/atom/config.cson`, which is a git repo which meant I got my settings back with `git checkout atom/config.cson` in that repo.

Here's my list:

```
[36mCommunity Packages (60)[39m /Users/plyons/.atom/packages
├── atom-format-shell@2.0.0
├── atom-htmltidy@5.1.0 (disabled)
├── atom-prettify@0.1.3
├── auto-detect-indentation@1.3.0
├── blame@0.12.0
├── busy-signal@1.4.3
├── change-case@0.6.5
├── custom-title@1.0.1
├── elm-format@3.0.0
├── elmjutsu@5.6.3
├── esformatter@2.3.0 (disabled)
├── expand-selection-to-quotes@0.11.0
├── formatter@2.12.4 (disabled)
├── formatter-tidy@1.0.1 (disabled)
├── git-plus@7.9.3
├── hard-wrap@1.1.0
├── hclfmt@1.2.0
├── html-to-elm@0.2.0
├── intentions@1.1.5
├── language-api-blueprint@1.0.2
├── language-applescript@0.3.0
├── language-docker@1.1.8
├── language-dockerfile@0.1.0
├── language-elm@1.5.0
├── language-hcl@0.4.0
├── language-jade@0.7.2
├── language-javascript-jsx@0.3.7
├── language-pug@0.0.21
├── language-rust@0.4.12
├── language-slim@0.6.0
├── language-sql-mysql@0.3.1
├── language-stylus@0.2.0
├── language-terraform@0.8.1
├── linter@2.2.0
├── linter-api-blueprint@0.4.0
├── linter-elm-make@0.24.3
├── linter-eslint@8.2.1
├── linter-js-yaml@1.2.7
├── linter-jsonlint@1.3.0
├── linter-rust@0.8.0
├── linter-shellcheck@1.4.6
├── linter-swagger@0.4.2
├── linter-terraform-syntax@1.2.1
├── linter-ui-default@1.6.8
├── linter-xmllint@1.4.3
├── mark@1.3.3
├── pane-by-type@1.0.1
├── pane-split-moves-tab@0.1.0
├── pipe@0.4.0
├── prettier-atom@0.38.0
├── pretty-json@1.6.4
├── rustfmt@0.1.3
├── sequential-command@0.3.0
├── shell-it@0.2.1
├── sort-lines@0.15.0
├── standard-formatter@2.8.0
├── terraform-fmt@0.3.0
├── toggle-quotes@1.0.1
├── wordcount@2.10.4 (disabled)
└── xml-formatter@0.11.0
```

**Comment by PointedEars** [comment]
Atom 1.20.0 x64 on Debian/Devuan GNU+Linux here.

I think there is strong indication that this bug is related to hyperclick or autoupdate-packages.

```
$ ls -ltr .atom/packages/ | tail -n 3
drwxr-xr-x  7 **** **** 4096 Sep 15 21:13 linter-ui-default
drwxr-xr-x  7 **** **** 4096 Sep 16 13:00 vim-mode-plus
drwxr-xr-x  8 **** **** 4096 Sep 17 01:00 js-hyperclick
```

Everything was OK yesterday.

As it happened previously, skins, Nuclide toolbars, text zoom, and shortcuts enabed/disabled state was reset to defaults today.

Apparently `js-hyperclick` was updated automatically at night by `autoupdate-packages`.

**Comment by iolsen** [comment]
What we hope is a fix has now shipped on the current stable (1.20.1) and beta (1.21.0-beta1) channels. Please let us know if you see this again after updating.

**Comment by ejpcmac** [comment]
It happened to me today on Atom 1.20.1 on Debian Linux after a `language-elixir` update. It has already happened a couple of times in the past six months —e.g. since I use Atom at work— sometimes after updates, but mainly just at startup without any obvious reason. I must notice that my computer has a slow hard drive, but this should be taken in account by Atom. It never happened to me on my Mac running on a SSD.

If this can help, follows my installed packages:

```
Community Packages (58) /home/***/.atom/packages
├── Sublime-Style-Column-Selection@1.7.4
├── atom-beautify@0.30.5
├── atom-clock@0.1.15
├── atom-elixir@0.2.3
├── atom-updater-linux@0.1.4
├── auto-detect-indentation@1.3.0
├── autoclose-html@0.23.0
├── autocomplete-clang@0.11.3
├── autocomplete-cmake@0.3.1
├── build@0.68.0
├── build-cmake@0.10.2
├── build-make@0.13.0
├── busy@0.7.0
├── busy-signal@1.4.3
├── editorconfig@2.2.2
├── file-icons@2.1.12
├── git-diff-details@1.4.0
├── git-time-machine@1.5.9
├── gitignore-snippets@0.2.5
├── highlight-selected@0.13.1
├── intentions@1.1.5
├── language-cmake@0.3.0
├── language-diff@0.7.0
├── language-doxygen@0.3.0
├── language-elixir@0.20.3
├── language-generic-config@1.4.0
├── language-gettext@0.6.1
├── language-gitignore@0.3.0
├── language-ini@1.19.0
├── language-latex@1.1.1
├── latex@0.47.0
├── latexer@0.3.0
├── linter@2.2.0
├── linter-chktex@1.3.1
├── linter-clang@4.1.2
├── linter-coffeelint@1.3.1
├── linter-elixirc@1.7.0
├── linter-mixed-indent@1.1.1
├── linter-pylint@2.1.0
├── linter-ruby@1.2.6
├── linter-ui-default@1.6.10
├── local-history@4.3.0
├── minimap@4.29.7
├── minimap-find-and-replace@4.5.2
├── minimap-git-diff@4.3.1
├── next-line-comment@0.2.1
├── pdf-view@0.59.0
├── phoenix-elixir-snippets@0.4.0
├── platformio-ide-terminal@2.7.0
├── process-palette@0.15.0
├── show-invisibles-plus@0.3.1
├── switch-header-source@0.22.1
├── sync-settings@0.8.3
├── theme-switch@0.5.0
├── todo-show@2.0.0
├── tool-bar@1.1.0
├── trailing-spaces@0.4.0
└── vim-mode-plus@1.8.0
```

**Comment by PointedEars** [comment]
Happened again :-1: 

```
$ atom -v
Atom    : 1.20.1
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0

$ find .atom/packages/* -type d ! -name . -prune -mtime -1 -printf '%t %f\n'
Thu Sep 28 22:45:26.8101827360 2017 language-babel
```

**Comment by adamlwgriffiths** [comment]
Boom, happened on OS-X 10.12.
So it's unlikely to be a file system race if it occurs across multiple OS.

`sync-settings` lost my git tokens.
`tool-bar` enabled itself
My overall colour theme is reset to the default
The welcome screen is showing up each start of Atom.
Atom asked me to select from a python parser (I already had one)

I have over 800gb free, so it's not a full disk issue.

```
$ atom --version
Atom    : 1.21.0
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

```
$ apm ls -i
Community Packages (38) ~/.atom/packages
├── atom-jsx-transform@0.2.0
├── atom-minify@0.8.0
├── autocomplete-python@1.10.2
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── editor-settings@1.1.3
├── git-control@0.9.0
├── highlight-selected@0.13.1
├── ink@0.7.4
├── intentions@1.1.5
├── language-lisp@0.2.0
├── language-restructuredtext@1.1.0
├── language-scheme@1.4.0
├── linter@2.2.0
├── linter-pep8@1.3.2
├── linter-pycodestyle@2.1.3
├── linter-pylint@2.1.0
├── linter-ui-default@1.6.8
├── lisp-paredit@0.5.5
├── markdown-scroll-sync@2.1.2
├── markdown-writer@2.7.2
├── minimap@4.29.6
├── monokai@0.24.0
├── multi-cursor@2.1.5
├── parinfer@1.20.0
├── pigments@0.40.2
├── pretty-json@1.6.4
├── proto-repl@1.4.21
├── proto-repl-charts@0.4.1
├── regex-railroad-diagram@0.19.3
├── robin-hood-syntax@1.0.0
├── rst-preview@1.3.1
├── set-syntax@0.3.2
├── sync-settings@0.8.3
├── todo-show@2.0.0
├── tool-bar@1.1.0
├── tree-ignore@0.6.2
└── vim-mode@0.66.0
```

**Comment by damieng** [comment]
Can anyone having this issue confirm if they had (when it happened) multiple Atom windows open?  Just trying to narrow down what might be the issue.

**Comment by geovanisouza92** [comment]
@damieng I was opening the first window of the day, just after starting up my machine and log in.

**Comment by focusaurus** [comment]
I almost always have several windows open so most likely when it happened to me I did.

**Comment by damieng** [comment]
Can anyone recall if their config.cson was deleted entirely, blanked out or contained a default atom config?

**Comment by damieng** [comment]
Likewise did anyone get a message about the config.json not being able to load?

If you get that message while Atom is running everything continue fine... until you reload whereby it helpfully tries to 'recover' by resetting your config.

Replacing that logic with one that contains a choice of "Edit your config" and "Reset your config" with the latter also backing the broken configuration up is possibly a good idea.

**Comment by iolsen** [comment]
Looking into this a bit on Friday I think I understand the problem. Three things, from most important to least:

1. We don’t do any cross-process synchronization of access to the settings file. It’s trivial to get interleaved reads/writes with multiple Atom windows open or side-by-side stable/beta installs. We need a lockfile (or 2) to ensure consistency: any number of readers is fine. Any single writer needs wait for all readers to finish. Any single reader needs to wait for the one and only writer to finish.
1. We expose the path to the settings file in the documented API (https://atom.io/docs/api/v1.21.1/Config#instance-getUserConfigPath), implying package authors can read/write it themselves, complicating the first issue. We should add a note to the docs saying not to do that and perhaps deprecate the method.
1. We needlessly reload the settings file in the same process where we just wrote it after calls to `set` or `unset`, making the first issue more severe. When we add locks, we should fix this, too.

**Comment by johncalvinyoung** [comment]
This or something like it happened to me today. Lost my installed packages and keybindings, possibly among other things.

**Comment by PointedEars** [comment]
Bug **confirmed** to **still exist in 1.22.1**

```
$ atom -v
Atom    : 1.22.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0

$ apm list --installed
Community Packages (172) /home/[…]/.atom/packages
├── MagicPython@1.0.12
├── Zen@0.18.0
├── atom-beautify@0.30.9
├── atom-css-unit-converter@1.1.1
├── atom-mysql-snippets@0.3.1
├── atom-php-navigation@0.6.0
├── atom-ternjs@0.18.3
├── atom-x@0.4.0
├── atomatigit@1.5.5
├── autoclose-html@0.23.0
├── autocomplete-awk@0.1.2
├── autocomplete-bibtex@1.1.0
├── autocomplete-clang@0.11.4
├── autocomplete-html-entities@0.2.0
├── autocomplete-paths@2.12.1
├── autocomplete-php@0.3.7
├── autocomplete-python@1.10.5
├── autocomplete-xml@0.9.4
├── autoupdate-packages@1.3.1
├── block-comment@0.4.3
├── busy-signal@1.4.3
├── character-table@0.4.3
├── clipboard-plus@0.5.1
├── color-picker@2.2.5
├── column-select@1.0.1
├── compare-files@0.8.1
├── css-snippets@1.1.0
├── ctrl-last-tab@0.3.0
├── data-atom@0.26.2
├── docblockr@0.13.4
├── emmet@2.4.3
├── emmet-snippets-compatibility@1.0.6
├── es6-javascript@1.0.0
├── ex-mode@0.18.0
├── filesize@3.0.0
├── fixmyjs@1.4.1
├── fold-comments@0.6.0
├── git-blame@1.4.0
├── git-clone@0.2.0
├── git-control@0.9.0
├── git-diff-details@1.4.0
├── git-difftool@0.2.7
├── git-go@1.0.5
├── git-hide@1.0.0
├── git-history@3.3.0
├── git-log@0.4.1
├── git-plus@7.10.0
├── git-rebase@0.1.1
├── git-status@0.3.4
├── git-time-machine@1.5.9
├── gitignore-snippets@0.2.5
├── goto@1.8.3
├── goto-definition@1.3.4
├── haskell-grammar@0.4.0
├── highlight-selected@0.13.1
├── hyperclick@0.1.5
├── hyperclick-php@0.2.1
├── ide-java@0.7.0
├── ide-php@0.7.4 (disabled)
├── ide-typescript@0.7.2
├── indent-toggle-on-paste@0.1.3
├── intentions@1.1.5
├── iso8601-timestamp@0.1.1
├── javascript-snippets@1.2.1
├── js-hyperclick@1.12.2
├── js-refactor@0.7.5
├── language-apache@1.8.0
├── language-awk@0.1.12
├── language-babel@2.81.0
├── language-diff@0.7.0
├── language-gettext@0.6.1
├── language-ini@1.19.0
├── language-latex@1.1.1
├── language-lua@0.9.11
├── language-sql-mysql@0.3.1
├── language-swift@0.5.0
├── language-tcltk@0.1.5
├── latex-autocomplete@1.1.1
├── latex-completions@0.3.3
├── latex-friend@0.0.11
├── latex-hyperclick@0.2.0
├── latexer@0.3.0
├── latextools@0.8.5
├── lcars-syntax@0.0.1
├── less-autocompile@1.0.0
├── less-than-slash@0.17.0
├── line-count-status@0.0.3
├── line-jumper@0.14.1
├── linter@2.2.0 (disabled)
├── linter-checkbashisms@0.1.5
├── linter-chktex@1.3.1
├── linter-csslint@2.0.0
├── linter-htmlhint@1.4.0
├── linter-jshint@3.1.6
├── linter-less@2.6.0
├── linter-mixed-indent@1.1.1
├── linter-mypy@2.4.3
├── linter-php@1.5.1
├── linter-pylint@2.1.0
├── linter-shellcheck@1.4.6
├── linter-ui-default@1.6.10
├── linter-xmllint@1.4.3
├── local-history@4.3.1
├── man@0.8.2
├── merge-conflicts@1.4.5
├── method-separator@0.2.0
├── minimap@4.29.7
├── minimap-cursorline@0.2.0
├── minimap-find-and-replace@4.5.2
├── minimap-git-diff@4.3.1
├── minimap-hide@0.3.0
├── minimap-highlight-selected@4.6.1
├── minimap-linter@2.1.3
├── minimap-pigments@0.2.2
├── minimap-split-diff@0.3.7
├── multi-cursor@2.1.5
├── multi-cursor-plus@1.2.0
├── multi-file-open@0.1.1
├── multirow-tabs@0.3.3
├── navigate@0.1.17
├── navigate-indent@0.1.0
├── next-occurrence@0.5.0
├── nuclide@0.270.0
├── nuclide-format-js@0.0.43
├── open-git-modified-files@0.2.4
├── open-recent@5.0.0
├── open-this@0.5.0
├── pdf-view@0.61.0
├── php-debug@0.2.6
├── php-extract-method@0.2.1
├── php-getters-setters@0.5.5
├── pigments@0.40.2
├── platformio-ide-terminal@2.7.0
├── preview@0.17.0
├── preview-inline@1.5.1
├── project-folder@1.5.0
├── project-view@0.6.2
├── promote-folder-to-project@2.0.2
├── punchclock@0.6.5
├── python-isort@0.0.7
├── python-tools@0.6.9
├── qolor@0.4.2
├── quantum-shell@0.7.1
├── quick-query@0.12.2
├── recent-projects@0.13.1
├── refactor@0.11.5
├── regex-tester@0.2.0
├── remember-file-positions@0.2.3
├── remote-edit@1.9.0
├── revert-buffer@0.6.0
├── script@3.18.0
├── set-syntax@0.4.0
├── seti-ui@1.9.0
├── show-origin@0.2.0
├── sloc@1.1.0
├── sort-lines@0.18.0
├── split-diff@1.5.1
├── switch-header-source@0.22.1
├── symbols-tree-nav@0.15.5
├── symbols-tree-view@0.14.0
├── term@0.2.4
├── text-manipulation@0.6.0
├── todo-show@2.1.0
├── tool-bar@1.1.0
├── tree-view-git-status@1.4.0
├── up2date@0.1.2
├── vim-mode-clipboard-plus@0.2.0
├── vim-mode-plus@1.24.1
├── w3c-validation@0.4.0
├── wordcount@2.10.4
├── xkcd-comics@0.3.0
└── xml-tools@0.2.1
```

**Comment by kylefox** [comment]
Just got bit by this as well 😐 Lost pretty much all of my settings. It's as if I installed Atom for the first time (seeing the welcome screen, telemetry consent screen, theme reset, etc).

**Comment by PointedEars** [comment]
@kylefox Which versions are you using?

**Comment by kylefox** [comment]
@PointedEars Sorry, should've mentioned that! I'm on `1.23.1`. I managed to restore my `~/.atom` folder using Time Machine, so everything is (thankfully) back to normal. But I'm not sure what caused this 🤔

**Comment by damieng** [comment]
We think the issue is multiple open copies of atom all trying to read and write the file at the same time.  Unfortunately we don't have a solid repro so it's a best-guess and synchronizing the writes is harder than it sounds due to the lack of cross-platform file lock facilities as well as the legacy fact that currently config changes are synchronous and any lock-based workaround would not be.

**Comment by geovanisouza92** [comment]
What about launching a "writer process" and making the others read-only?

Another approach could be build some kind of "consensus protocol" (I'm thinking on raft, but simpler), making multiple processes agree what of it can write on a given moment.

**Comment by PointedEars** [comment]
@damieng Did you mean it is _asynchronous_ now (possibility of everyone writing at the same time) but would not be then?

**Comment by kylefox** [comment]
@damieng OH — that's interesting, because I also noticed (but forgot to mention) that I saw _two_ Atom icons in my task switcher when this first happened! One of them would correctly bring Atom into focus, the other appeared to do nothing.

IIRC, the double-icon issue persisted even after closing & re-launching Atom. I ended up restarting my computer and the issue went away.

**Comment by damieng** [comment]
@PointedEars I mean that the call from Atom to the file system for read/write of the config is currently synchronous -i.e. it can block threads. Adding locking to that would make it worse so ideally it would be async.  If you scroll up you can see I spent a lot of time trying to go down that rabbit hole but the failure of so many tests plus the difficulty in writing tests to prove this was any better meant I lost confidence in that approach.

One option I've been thinking about is having only one of each Atom's threads read and write to the config. Right now every Atom window has both the UI and the main process and each of them read/write to the config. Having one marshal changes and requests over to the other would reduce the amount of processes trying to access that file by 50% so that might be a change worth trying. 

Another thing I've been considering is having the config copy the file to a .bak periodically when it writes the config system.  

Thoughts welcome.

**Comment by damieng** [comment]
@geovanisouza92 If it were shared between copies of Atom that would probably work quite well although it does introduce a whole separate issue of ensuring there is only one and that it communicates correctly with all instances of Atom - difficult when you can run multiple different versions of Atom with the same config file.

**Comment by pbarbiero** [comment]
This just happened to me on Atom 1.23.3

**Comment by maxbrunsfeld** [comment]
Hi everyone, sorry about this. The fix will ship in Atom 1.25 (or 1.25 Beta, if you're so inclined).

**Comment by krrishh-org** [comment]
Using atom1.25.0-beta0 

```
Installed Community Packages (16) 
├── activate-power-mode@2.7.0
├── atom-beautify@0.30.9
├── atom-ide-ui@0.9.1
├── busy-signal@1.4.3
├── caniuse@0.9.2
├── file-icons@2.1.16
├── ide-typescript@0.7.3
├── intentions@1.1.5
├── language-docker@1.1.8
├── linter@2.2.0
├── linter-docker@0.3.0
├── linter-eslint@8.4.1
├── linter-ui-default@1.6.10
├── sort-lines@0.18.0
├── split-diff@1.5.2
└── teletype@0.8.0
```
Clears config.cson **everytime** I restart atom.

**Comment by Ben3eeE** [comment]
@krrishh-org An additional fix was included in Atom 1.25.0-beta2. Do you still experience this issue if you update to the latest beta release?

**Comment by dominic-p** [comment]
For what it's worth, I'm seeing this on 1.24.0 on Windows 10. I can reliably get the error below by clicking on the "Open Config Folder" button on the settings screen.

```
Failed to load config.cson
[stdin]:1:1: error: unexpected

[bunch of white space here]

^
```

I haven't tried the beta though.

**Comment by krrishh-org** [comment]
@Ben3eeE, It seems to be working fine after updating to 1.25.0-beta2.  Thank you.

**Comment by Ben3eeE** [comment]
@krrishh-org That's great to hear! Thank you for reporting back :bowing_man:

**Comment by ernest33** [comment]
@krrishh-org @Ben3eeE  Nope i got this issue this morning using 1.25.0-beta2 x64 :-/
Currently i keep a safe copy of my config.cson :-)

**Comment by PointedEars** [comment]
@ernest33 Too bad. But as @ungb and @adamlwgriffiths mentioned, for the workaround you only need [sync-settings](https://atom.io/packages/sync-settings) and keep your config gist up-to-date. WFM.

**Comment by krrishh-org** [comment]
> @krrishh-org That's great to hear! Thank you for reporting back :bowing_man:

It happened again. this time on 1.25.0-beta2. It was working fine until now and after I restarted it now, I lost the settings again. The difference was this time I launched atom from terminal as follows:

```
$ atom-beta ./
```

and also the reason for restart was: the editor crashed after I ran `npm install` in the project folder and it showed a pop-up with options 'Close Editor', 'Reload Editor', 'Wait'. I chose 'Reload Editor'. It reloaded the editor fine with all the settings intact. But during next restart it overwrote the `config.cson` file

**Comment by rsese** [comment]
Re-opening since folks are hitting this in `1.25.0-beta2` - @ernest33 @krrishh-org do either of you have consistent steps to reproduce that you can share?

**Comment by krrishh-org** [comment]
Yes,

Steps to reproduce:

- Open atom editor
-  Open command palette `Ctrl+Shift+P`
- find and trigger Window:Reload   (or use the shortcut `Ctrl+Shift+F5`)
(refreshes the editor normally with the settings still intact)
- Close atom and reopen atom and :boom:  my settings are gone

**Comment by Ben3eeE** [comment]
@krrishh-org Do you see this if you start Atom in safe mode `atom --safe`? If not can you provide the list of packages you have installed `apm list`?

**Comment by krrishh-org** [comment]
No. I don't see this in safe mode `atom-beta --safe`.

Here is output of `apm-beta list`
```
Built-in Atom Packages (92)
├── atom-dark-syntax@0.29.0
├── atom-dark-ui@0.53.1
├── atom-light-syntax@0.29.0
├── atom-light-ui@0.46.1
├── base16-tomorrow-dark-theme@1.5.0
├── base16-tomorrow-light-theme@1.5.0
├── one-dark-ui@1.10.10
├── one-light-ui@1.10.10
├── one-dark-syntax@1.8.2
├── one-light-syntax@1.8.2
├── solarized-dark-syntax@1.1.4
├── solarized-light-syntax@1.1.4
├── about@1.8.0
├── archive-view@0.64.2
├── autocomplete-atom-api@0.10.7
├── autocomplete-css@0.17.5
├── autocomplete-html@0.8.4
├── autocomplete-plus@2.40.2
├── autocomplete-snippets@1.12.0
├── autoflow@0.29.3
├── autosave@0.24.6
├── background-tips@0.27.1
├── bookmarks@0.45.1
├── bracket-matcher@0.89.1
├── command-palette@0.43.3
├── dalek@0.2.1
├── deprecation-cop@0.56.9
├── dev-live-reload@0.48.1
├── encoding-selector@0.23.8
├── exception-reporting@0.43.1
├── find-and-replace@0.215.5
├── fuzzy-finder@1.7.5
├── github@0.10.1
├── git-diff@1.3.9
├── go-to-line@0.33.0
├── grammar-selector@0.49.9
├── image-view@0.62.4
├── incompatible-packages@0.27.3
├── keybinding-resolver@0.38.1
├── line-ending-selector@0.7.5
├── link@0.31.4
├── markdown-preview@0.159.20
├── metrics@1.2.6
├── notifications@0.70.2
├── open-on-github@1.3.1
├── package-generator@1.3.0
├── settings-view@0.254.0
├── snippets@1.3.1
├── spell-check@0.72.7
├── status-bar@1.8.15
├── styleguide@0.49.10
├── symbols-view@0.118.2
├── tabs@0.109.1
├── timecop@0.36.2
├── tree-view@0.221.3
├── update-package-dependencies@0.13.1
├── welcome@0.36.6
├── whitespace@0.37.5
├── wrap-guide@0.40.3
├── language-c@0.59.2
├── language-clojure@0.22.7
├── language-coffee-script@0.49.3
├── language-csharp@1.0.1
├── language-css@0.42.10
├── language-gfm@0.90.3
├── language-git@0.19.1
├── language-go@0.45.2
├── language-html@0.49.0
├── language-hyperlink@0.16.3
├── language-java@0.28.0
├── language-javascript@0.128.3
├── language-json@0.19.1
├── language-less@0.34.2
├── language-make@0.22.3
├── language-mustache@0.14.5
├── language-objective-c@0.15.1
├── language-perl@0.38.1
├── language-php@0.43.1
├── language-property-list@0.9.1
├── language-python@0.49.2
├── language-ruby@0.71.4
├── language-ruby-on-rails@0.25.3
├── language-sass@0.61.4
├── language-shellscript@0.26.1
├── language-source@0.9.0
├── language-sql@0.25.10
├── language-text@0.7.3
├── language-todo@0.29.4
├── language-toml@0.18.2
├── language-typescript@0.3.2
├── language-xml@0.35.2
└── language-yaml@0.31.2

Community Packages (15) /home/gopi/.atom/packages
├── activate-power-mode@2.7.0
├── atom-beautify@0.30.9
├── atom-ide-ui@0.9.2
├── busy-signal@1.4.3
├── caniuse@0.9.2
├── file-icons@2.1.17
├── ide-typescript@0.7.5
├── intentions@1.1.5
├── language-docker@1.1.8
├── linter@2.2.0
├── linter-docker@0.3.0
├── linter-eslint@8.4.1
├── linter-ui-default@1.6.10
├── sort-lines@0.18.0
└── split-diff@1.5.2
```

**Comment by Ben3eeE** [comment]
@krrishh-org Additionally do you have anything in your init script? You can open it using `application:open-your-init-script` from the command palette.

**Comment by krrishh-org** [comment]
@Ben3eeE  Nothing. It is empty (except for some commented instructions).

**Comment by maxbrunsfeld** [comment]
@krrishh-org Thanks for the info. that is really helpful. It looks like your list of `Community Packages` is not too long. Any chance you'd be interested in trying to narrow down which package(s) are causing the problem? Like, disable half of your community packages and try to repro again?

**Comment by Arcanemagus** [comment]
If it helps, the commonality with my package list is:
```
├── atom-beautify@0.30.9
├── atom-ide-ui@0.9.2
├── busy-signal@1.4.3
├── ide-typescript@0.7.5
├── intentions@1.1.5
├── language-docker@1.1.8
├── linter@2.2.0
├── linter-docker@0.3.0
├── linter-eslint@8.4.1
├── linter-ui-default@1.6.10
├── sort-lines@0.18.0
```

Since you have `atom-ide-ui` and `linter` installed, this _may_ also depend on what file(s) you have open at the time Atom launches.

**Comment by krrishh-org** [comment]
@maxbrunsfeld I have narrowed it down to `linter-ui-default@1.6.10`. But after the updating to [linter-ui-default@1.7.0](https://github.com/steelbrain/linter-ui-default/releases/tag/v1.7.0) (released 11 hours ago) this issue seems to disappear. Can anyone else confirm that? Thank you.

**Comment by delfuego** [comment]
I updated from linter-ui-default@1.6.10 to linter-ui-default@1.7.1 — and my settings were reset. So, I'm not sure if linter has fixed the problem.

(Thankfully, I had been watching this thread, and had saved my config.cson file before starting the update... so it was easy to remedy after the fact.)

**Comment by delfuego** [comment]
Oh, and of note, I'm running Atom 1.25.0-beta2, and the only other package that updated during the linter-ui-default update was atom-beautify. So it really does look like linter-ui-update is the culprit here, or at least *a* culprit.

**Comment by Arcanemagus** [comment]
`linter-ui-default` has the possibility of calling `atom.config.set` in it's `deactivate` [here](https://github.com/steelbrain/linter-ui-default/blob/f0a62d36393d8a29ad230e22b1c7505dff1284e5/lib/panel/dock.js#L120), I think that's the only relation to this issue (v1.7.1 may have made this less common, but didn't _change_ that in any real way, so I'm not surprised you are still seeing this).

It sounds like there might still be a race condition when shutting down Atom.

**Comment by ukdor** [comment]
I just had this happen to me.  I saw an update to 1 package available "linter-ui-default" version 1.17.0.  I chose to update it.  I had 2 Atom windows open on my Windows 7 64-bit system.  I closed the window in which I didn't click "update" and then eventually closed the window running the update.  Once I restarted I got errors about an issue with the Seti UI package not be compatible and my theme and syntax were reset.  As I looked closer, I noticed other packages settings were reset.  My core setting to DISABLE auto-update (I have to stay on version 1.19 due to corporate security that prevents updates until certified) of Atom was back to enabled, etc.

Is there a method to backup configurations?  I'll have to search for that and keep it handy.

My package list:
├── apex-syntax@0.2.0
├── apex-ui@1.0.3
├── atom-live-server@2.2.0
├── busy-signal@1.4.3
├── city-lights-syntax@1.1.8
├── city-lights-ui@1.3.3
├── dark-bint-syntax@0.10.0
├── dark-bint-ui@0.5.2
├── expose@0.15.0
├── file-icons@2.1.15
├── git-checkout@0.3.2
├── gitlab-integration@0.4.6
├── hide-files@0.3.0
├── highlight-selected@0.13.1
├── intentions@1.1.5
├── jackhammer-syntax@1.1.0
├── language-awk@0.1.12
├── language-chef@0.11.0
├── language-dtd@0.2.1
├── language-powershell@4.0.0
├── language-unix-shell@0.40.1
├── language-xst@0.3.3
├── linter@2.2.0
├── linter-ruby@1.2.6
├── linter-tidy@2.3.1
├── linter-ui-default@1.7.1
├── markdown-pdf@2.1.0
├── markdown-preview-plus@2.4.16
├── markdown-preview-plus-opener@0.2.0
├── markdown-scroll-sync@2.1.2
├── mdtable@0.0.1
├── merge-conflicts@1.4.5
├── minimap@4.29.8
├── org-mode@0.3.11
├── pigments@0.40.2
├── pristine-ui@1.1.11
├── pure-syntax@0.3.0
├── rest-client@1.3.1
├── seti-classic@0.10.3
├── seti-classic-syntax@0.5.0
├── seti-icons@1.5.4
├── seti-syntax@1.1.3
├── seti-ui@1.10.0
├── spacegray-atom-dark-syntax@1.3.2
├── spacegray-atom-dark-ui@1.0.2
├── Sublime-Style-Column-Selection@1.7.4
└── tree-view-git-status@1.4.0

**Comment by mattmattmatt** [comment]
I also just experienced settings loss. It did not seem to happen during a package update but while running the specs for the `atom/github` package. 

It was mid-development and not during or after a restart of Atom. Suddenly my color theme changed from light to dark, and after force-quitting Atom and restarting, all the Welcome panes etc were up.

`apm ls -i`:
```bash
├── atom-beautify@0.32.2
├── atom-json-color@0.6.0
├── atom-wrap-in-tag@0.6.0
├── build@0.70.0
├── build-make@0.13.0
├── busy@0.7.0
├── busy-signal@1.4.3
├── cucumber@0.6.0
├── custom-invisibles@0.1.2
├── emmet-simplified@2.5.0
├── esformatter@2.3.0
├── file-icons@2.1.17
├── git-blame@1.6.0
├── git-time-machine@1.5.9
├── highlight-selected@0.13.1
├── indent-guide-improved@1.4.13
├── intentions@1.1.5
├── keyboard-localization@1.5.0
├── language-babel@2.84.0
├── language-groovy@0.7.0
├── language-lua@0.9.11
├── linter@2.2.0
├── linter-eslint@8.4.1
├── linter-js-standard-jsx@4.1.2
├── linter-stylelint@4.2.0
├── linter-ui-default@1.7.1
├── markdown-preview-opener@0.1.1
├── merge-conflicts@1.4.5
├── minimap@4.29.8
├── minimap-cursorline@0.2.0
├── minimap-find-and-replace@4.5.2
├── minimap-highlight-selected@4.6.1
├── minimap-linter@2.1.3
├── pigments@0.40.2
├── prettier-atom@0.52.0
├── react@0.18.0
├── react-function-to-class@0.2.3
├── regex-railroad-diagram@0.19.4
├── svg-preview@0.12.1
├── svgo@3.0.0
├── sync-settings@0.8.5
├── teletype@0.10.0
├── terminal-tab@0.5.4
├── tidy-tabs@0.4.0
├── webbox-color@0.5.9
└── zentabs@0.8.9
```

**Comment by mattmattmatt** [comment]
I experienced the same settings loss behavior again while running the specs for the `atom/github´ package. This time I had DevTools in the spec runner open and copied [all information into a gist](https://gist.github.com/mattmattmatt/c7bd10f671903c4d791283b2e0e9e1d1):  
https://gist.github.com/mattmattmatt/c7bd10f671903c4d791283b2e0e9e1d1

[Here is a branch](https://github.com/mattmattmatt/github/commit/81de9167d4faf272e810928e8c469d2101e74322) with the state of the package when this happens:  
https://github.com/mattmattmatt/github/commit/81de9167d4faf272e810928e8c469d2101e74322

Hope that helps.

**Comment by fuyutsuki** [comment]
I have also encountered the same problem. I hope the following information contributes to the early solution.
Also, in my case, correctly started by clicking "`Open with Atom`" after modifying config.cson, but clicking "`Atom.exe`" (or its shortcut) A white screen appeared and it **freezed**.
`apm ls -i`:
```bash
Community Packages (40) C:\Users\user\.atom\packages
├── atom-autocomplete-php@0.25.6
├── atom-bootstrap3@1.2.12
├── autocomplete-java@1.2.7
├── autocomplete-json@5.5.2
├── autocomplete-paths@2.12.2
├── busy-signal@1.4.3
├── color-picker@2.2.5
├── convert-to-utf8@0.2.4
├── docblockr@0.13.7
├── duotone-dark-sea-syntax@2.1.0
├── editorconfig@2.2.2
├── file-icons@2.1.17
├── git-time-machine@1.5.9
├── highlight-selected@0.13.1
├── hyperclick@0.1.5
├── intentions@1.1.5
├── japanese-menu@1.14.0
├── language-batch@0.4.0
├── language-gitignore@0.3.0
├── language-ini@1.19.0
├── language-kotlin@0.5.0
├── language-lua@0.9.11
├── language-powershell@4.0.0
├── linter@2.2.0
├── linter-jsonlint@1.3.0
├── linter-php@1.5.1
├── linter-sass-lint@1.8.3
├── linter-ui-default@1.7.1
├── minimap@4.29.8
├── minimap-bookmarks@0.4.2
├── minimap-find-and-replace@4.5.2
├── minimap-git-diff@4.3.1
├── minimap-highlight-selected@4.6.1
├── minimap-linter@2.1.3
├── minimap-selection@4.5.0
├── platformio-ide-terminal@2.8.1
├── pristine-ui@1.1.11
├── seti-ui@1.10.0
├── sort-lines@0.18.0
└── teletype@0.10.0
```

`atom --version`:
```bash
Atom    : 1.26.0-beta0
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

`apm --version`:
```bash
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.26.0-beta0
python 2.7.14
git 2.16.0.windows.2
visual studio 2015
```

**Comment by tylerFowler** [comment]
Same thing here.. first noticed this a few weeks ago but figured it was some fluke thing on my end but it's happened a few times since then, has actually happened both yesterday & today. Below is some environment information, worth noting that for a while I've been starting atom via `atom -d .` because I'm using an in development package.

`apm ls -i`:
```
[36mCommunity Packages (33)[39m ~/.atom/packages
├── Visual-Studio-2015-Dark-Theme@2.4.2
├── alinement@1.5.0
├── atom-clock@0.1.16
├── atom-ide-ui@0.9.4
├── atom-jinja2@0.6.0
├── atom-spotify2@2.1.2
├── atom-ternjs@0.18.3
├── base16-papercolor-dark-syntax@1.0.0
├── base16-papercolor-light-syntax@1.1.0
├── busy-signal@1.4.3
├── duotone-dark-space-syntax@2.1.0
├── duotone-dark-syntax@2.1.0
├── duotone-light-syntax@2.1.0
├── file-icons@2.1.18
├── git-plus@7.10.2
├── go-plus@5.8.2
├── go-signature-statusbar@1.2.3
├── gruvbox-plus-syntax@1.3.1
├── hyperclick@0.1.5
├── ide-java@0.8.2
├── intentions@1.1.5
├── language-docker@1.1.8
├── language-gitignore@0.3.0
├── language-groovy@0.7.0
├── language-javascript-jsx@0.3.7
├── language-systemd@0.1.1
├── language-terraform@0.8.1
├── linter@2.2.0 (disabled)
├── linter-docker@0.3.0
├── linter-eslint@8.4.1
├── linter-ui-default@1.7.1
├── redmond-syntax@1.0.2
└── vim-mode-plus@1.31.1
```

`atom --version`:
```
Atom    : 1.26.0-beta0
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

`apm --version`:
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.26.0-beta0
python 2.7.10
git 2.16.2
```

**Comment by j-sm-n** [comment]
The same issue just happened to me for the first time today. It happened again after I updated Hyperclick.

`apm ls -i`:
```
Community Packages (42) /Users/admin/.atom/packages
├── 3.0.0
├── aligner@1.2.4
├── aligner-css@1.2.1
├── aligner-javascript@1.3.0
├── aligner-php@1.2.0
├── apex-syntax@0.2.0
├── ariake-dark-syntax@0.1.3
├── atom-ide-ui@0.9.4
├── atom-material-syntax@1.0.8
├── atom-material-ui@2.1.3
├── autocomplete-js-import@1.4.0
├── autocomplete-php@0.3.7
├── busy-signal@1.4.3
├── docblockr@0.13.7
├── file-icons@2.1.18
├── git-time-machine@1.5.9
├── goto-definition@1.3.4
├── hyperclick@0.1.5
├── hyperclick-php@0.2.1
├── intentions@1.1.5
├── language-babel@2.84.0
├── linter@2.2.0 (disabled)
├── linter-eslint@8.4.1
├── linter-php@1.5.1
├── linter-ui-default@1.7.1
├── node-debugger@1.10.1
├── php-debug@0.2.6
├── php-integrator-annotations@1.2.0
├── php-integrator-base@3.2.1
├── php-integrator-call-tips-legacy-php56@1.2.3
├── php-integrator-navigation@1.2.1
├── php-integrator-refactoring@1.4.1
├── php-integrator-tooltips-legacy-php56@1.1.6
├── php-twig@4.0.0
├── pigments@0.40.2
├── platformio-ide-terminal@2.8.1
├── project-manager@3.3.5
├── propel-snippets@0.12.0
├── pure-syntax@0.3.0
├── qolor@0.4.2
├── react-snippets@1.0.1
└── todo-show@2.2.0
```
`atom --version`
```
Atom    : 1.25.0
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```
`apm --version`
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.25.0
python 2.7.10
git 2.14.3
```

**Comment by niaodan2b** [comment]
same issue here...I opened Atom this morning and saw the welcome screen, with my settings all gone. I'm using Windows10(1709).

`apm ls -i`
```
Community Packages (15) C:\Users\niaodan\.atom\packages
├── atom-prettify@0.1.3
├── atom-ternjs@0.18.3
├── busy-signal@1.4.3
├── element-helper@0.5.4
├── file-icons@2.1.18
├── highlight-selected@0.13.1
├── indent-detective@0.2.0
├── intentions@1.1.5
├── language-restructuredtext@1.1.0
├── language-vue@0.23.1
├── linter@2.2.0
├── linter-jshint@3.1.6
├── linter-ui-default@1.7.1
├── minimap@4.29.8
└── pretty-json@2.0.0
```
`atom --version`
```
Atom    : 1.25.0
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
```
`apm --version`
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.25.0
python
git 2.16.2.windows.1
visual studio 2015
```

**Comment by smashwilson** [comment]
> I also just experienced settings loss. It did not seem to happen during a package update but while running the specs for the atom/github package.

I've tracked this one down to #17040. There's a chance that the other cases we're seeing share the same root cause, but I'm leery of declaring victory just yet.

I think I'm also going to work on a mitigation that backs up your config file to lessen the pain until we can really have confidence here.

**Comment by Arcanemagus** [comment]
I'm going to close this issue as it has had several different problems brought together in it, and at least one of them has been solved already.

Thanks to the reports of everyone here a commonality between all the reports was determined and has been filed as #17060, feel free to follow that issue if you want further updates on this problem.

Thanks again for all of your reports!

**Comment by lparsons** [comment]
I just ran into this issue, not really sure what, if anything was updated. Ubuntu 16.04.

`apm ls -i`
```
Community Packages (28) /home/lparsons/.atom/packages
├── Sublime-Style-Column-Selection@1.7.4
├── atom-beautify@0.32.5
├── atom-ide-ui@0.12.0
├── atom-jinja2@0.6.0
├── autocomplete-python@1.10.5
├── busy-signal@1.4.3
├── editorconfig@2.2.2
├── git-plus@7.10.2
├── ide-python@1.0.0
├── intentions@1.1.5
├── language-batch@0.4.0
├── language-matlab@0.2.1
├── language-r@0.4.2
├── language-tcl@0.1.2
├── linter@2.2.0
├── linter-flake8@2.3.0
├── linter-htmlhint@1.4.0
├── linter-markdown@5.2.2
├── linter-mixed-indent@1.1.1
├── linter-perlcritic@2.1.0
├── linter-ui-default@1.7.1
├── linter-xmllint@1.4.4
├── markdown-writer@2.7.3
├── minimap@4.29.8
├── python-indent@1.1.5
├── python-tools@0.6.9
├── sync-settings@0.8.6
└── tablr@1.8.3
```

`atom --version`
```
Atom    : 1.27.2
Electron: 1.7.15
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

`apm --version`
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.27.2
python 3.6.5
git 2.7.4
```

Any tips on restoring?

**Comment by pstanton** [comment]
This happens to me every so often. I'm currently on 1.28.1 however I think I upgraded since the last incident, so this is purely anecdotal. Problem is, the last time it happened not only did i get the welcome screen, lose my settings etc, I also lost the ability to use the embedded terminal.

Reading up on similar issues points to a node/nuclide version incompatibility, however neither have changed since the previous successful case. 
I get:

> Error starting process:
> Error: A dynamic link library (DLL) initialization routine failed.
> \\?\C:\Users\pstanton\.atom\packages\nuclide\node_modules\nuclide-prebuilt-libs\pty\build\pty-v0.1.4-node-v57-win32-x64\pty.node

I have lost hours trying to rectify this, and now have reverted back to a native shell and am considering dropping atom altogether.

**Comment by andrew-aladev** [comment]
@lparsons, @pstanton I can offer to use local git repository in `.atom` folder. If something is broken - you can always checkout previous versions of any config. I am reproducing this issue 2-3 times per month.

**Comment by mslinn** [comment]
I hit this problem today with v1.29.0 on Windows 10. Running this command solved it:
```
apm remove linter-solium
```

**Comment by jrial** [comment]
@mslinn that's good to know, but as you can see, it's not the only candidate. I'm the one who reported this, and I never had that package installed.

Also, how do you know removing that package fixed it? The bug doesn't manifest itself consistently. It's hard to determine if the issue is really gone since we don't yet know what triggers it, and since it manifests itself rarely and for all intents and purposes, at this moment, randomly.

**Comment by mslinn** [comment]
@jrial Yes, everything you say is true. The problem did come back until I also removed `rust-api-docs-helper`. After many restarts the problem has not manifested again.

**Comment by lparsons** [comment]
This is becoming an increasingly problematic issue. Didn't happen for a while, but today I ran into it again.

`apm ls -i`
```
Community Packages (35) /home/lparsons/.atom/packages
├── Sublime-Style-Column-Selection@1.7.4
├── atom-beautify@0.33.1
├── atom-ide-ui@0.13.0
├── atom-jinja2@0.6.0
├── autocomplete-python@1.10.12
├── busy-signal@1.4.3
├── editorconfig@2.2.2
├── file-icons@2.1.24
├── git-plus@8.1.0
├── ide-python@1.0.0
├── intentions@1.1.5
├── language-batch@0.4.0
├── language-crontab@1.0.0
├── language-matlab@0.2.1
├── language-r@0.4.2
├── language-restructuredtext@1.1.2
├── language-tcl@0.1.2
├── language-tt@0.5.1
├── linter@2.2.0
├── linter-flake8@2.3.0
├── linter-htmlhint@1.5.1
├── linter-markdown@5.2.2
├── linter-mixed-indent@1.1.1
├── linter-perlcritic@2.1.0
├── linter-ui-default@1.7.1
├── linter-xmllint@1.4.4
├── markdown-writer@2.9.0
├── minimap@4.29.8
├── monokai@0.24.0
├── python-indent@1.1.6
├── python-tools@0.6.9
├── seti-syntax@1.2.0
├── seti-ui@1.11.0
├── sync-settings@0.8.6
└── tablr@1.8.3
```
`atom --version`
```
Atom    : 1.30.0
Electron: 2.0.5
Chrome  : 61.0.3163.100
Node    : 8.9.3
```
`apm --version`
```
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.30.0
python 3.6.6
git 2.7.4
```

**Comment by e7o-de** [comment]
Stumbled over that issue today. Very annoying ... Thought I could solve it by just deleting all of my config (rm of ~/.atom), but it's still not working. Info:

```
box ~ $ apm-beta ls -i
Community Packages (0) /home/sven/.atom/packages
└── (empty)
```

```
box ~ $ atom-beta -v
Atom    : 1.31.0-beta0
Electron: 2.0.7
Chrome  : 61.0.3163.100
Node    : 8.9.3
```

```
box ~ $ apm-beta -v
apm  2.1.0
npm  6.2.0
node 8.9.3 x64
atom 1.31.0-beta0
python 2.7.15rc1
git 2.17.1
```

Absolutely no idea what this could be, it feels a bit like the system around Atom is broken, like npm or so, by installing a broken package there ...

**Comment by e7o-de** [comment]
Played a bit. "Solved" it somehow partially by installing atom instead of atom-beta. Not every setting gets stored at first try, but after configuring it twice everything works. After I fixed it with atom, it sometimes even works with atom-beta (same profile) for one or two launches ...

Entirely killed ~/.atom and ~/.config/Atom/ -- what else is still there where it could store settings? I mean, it has obviously some effect on atom-beta when running atom ...

**Comment by rsese** [comment]
For anyone that's still seeing this problem, can you open a new issue and fill out the issue template with as much information as you can so the team can track it?

**Comment by bradleyy1012** [comment]
i guess im not alone D'=

**Comment by UziTech** [comment]
looks like it may be an issue when calling `atom.config.set` when selector is `"*"`

```js
atom.config.set("test.test", true, {scopeSelector: "*"});
```

running this in debug console will wipe all settings under "*" scope and add test.test setting

**Comment by bonovski** [comment]
This is super annoying. 
How can the config.cson be completely blank after startup?

**Comment by lparsons** [comment]
Just happened to me again. Another possibly related issue? https://github.com/atom/atom/issues/17060

**Comment by nelson6e65** [comment]
Maybe `linter`/`linter-ui-default` has to do with this? I get a meesage that "Atom needs to restart to apply..." on startup.

**Comment by Azovsky** [comment]
I still have this problem periodically on Win64 8.1.
e.g. just updated Atom 1.36.0 to 1.36.1 and have "clear config" again!

**Comment by aalok-sathe** [comment]
I get this on every reboot, and occasionally also mid-session. Faced on MacOS X, but not on Ubuntu 16.04.

**Comment by Moustachey** [comment]
I'm having this issue weekly and it's frustrating it's still happening after a simple package update. I'll follow this issue but I'll use VS Code until this is resolved.

**Comment by nelson6e65** [comment]
https://github.com/atom/atom/issues/14909#issuecomment-467491087
> I disabled to see if this error happens again. I had the `linter` disabled, because of `atom-ide-ui`, but I forgot to disable `linter-ui-default`.

@Moustachey I disabled `linter-ui-default` and seems to be solved. Don't know why, but... If you have `atom-ide-ui` you can uninstall use it instead of that.

**Comment by PointedEars** [comment]
> I'm having this issue weekly and it's frustrating it's still happening after a simple package update. I'll follow this issue but I'll use VS Code until this is resolved.

With the `sync-settings` package installed there is really no reason to abandon Atom (for VS Code) because of this.

**Comment by rsese** [comment]
Really sorry folks are still hitting this - what would be helpful is if you can reproduce, please open a new issue with the template filled out so we can take a look.  What would extra helpful is exact steps to reproduce if possible though I understand in this particular case it's not straightforward.

**Comment by Moustachey** [comment]
> With the `sync-settings` package installed there is really no reason to abandon Atom (for VS Code) because of this.
Thanks @PointedEars I'll give that a try!

@rsese I get this issue randomly after updating a package and restarting Atom. If I experience the issue again and they're different reproduction steps I'll file a separate issue.
